### PR TITLE
add required datacenter param to all calls which may use get_vm_interface

### PR DESCRIPTION
**Pull request by scambra**
Calling server.interfaces.get('mac_address') failed because datacenter is required
It could fail calling server.update_vm_interface because datacenter is required,
or calling interface.destroy

### CODE REVIEW

**Comment by chris1984** [comment]
@scambra thanks I will start testing this.

File: lib/fog/vsphere/models/compute/server.rb [hunk 1]
```diff
@@ -230,6 +230,7 @@ def add_interface(attrs)
 
         def update_interface(attrs)
           wait_for { !ready? } if interface_ready? attrs
+          attrs = attrs.merge(datacenter: datacenter) unless attrs.key? :datacenter
```
Comment (timogoebel) [inline lib/fog/vsphere/models/compute/server.rb]: You can use `attrs[:datacenter] = datacenter` here. That avoids the call to `merge`.

Comment (scambra) [reply lib/fog/vsphere/models/compute/server.rb]: changed

**Comment by chris1984** [comment]
Thanks, @scambra for all the contributions.

**Pull request merged**